app/models/account.rb

```diff
@@ -6,9 +6,10 @@ class Account < ApplicationRecord
 
   broadcasts_refreshes
   belongs_to :family
-  has_many :balances
-  has_many :valuations
-  has_many :transactions
+
+  has_many :balances, dependent: :destroy
+  has_many :valuations, dependent: :destroy
+  has_many :transactions, dependent: :destroy
 
```

Comment:
This is correct behavior and matches the behavior at the database level.  The previous version was an oversight.